ngraph_bridge/ngraph_encapsulate_impl.h

```diff
@@ -0,0 +1,220 @@
+/*******************************************************************************
+ * Copyright 2017-2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#ifndef NGRAPH_TF_ENCAPSULATE_IMPL_H_
+#define NGRAPH_TF_ENCAPSULATE_IMPL_H_
+
+#include <ostream>
+#include <vector>
+
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/graph/graph.h"
+
+#include "ngraph/ngraph.hpp"
+
+#include "logging/ngraph_log.h"
+#include "ngraph_bridge/ngraph_freshness_tracker.h"
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+using NgFunctionIOCache = std::unordered_map<
+    std::shared_ptr<ngraph::runtime::Executable>,
+    std::vector<std::pair<void*, shared_ptr<ng::runtime::Tensor>>>>;
+
+class NGraphEncapsulateImpl {
+ public:
+  // Ngraph Encapsulate Implementation class for EncapsulateOp class
+  explicit NGraphEncapsulateImpl(string name);
+
+  // Get tensorflow input tensors, input shapes, static_inputs to Compute
+  // Signature
+  Status ComputeSignature(const std::vector<Tensor>& tf_input_tensors,
```

Comment:
Make all pf these methods private except for the ones that are to be called by there classes.

Reply:
General comments - applies to all code:
1. Add adequeate comments to each function - especially any underlying assumption
2. Please follow one coding style - which is most likely snake_case (double check with @shresthamalik.
3. No public data or unnecessary methods.

Reply:
@avijit-nervana  All our other files/classes(MarkForClustering, FreshnessTracker, BackenManager) follow TF's convention which follows camelCase (but with the first alphabet also capital). So I am recommending moving all the getter/setters to that style.